File: Kotlin/Solucionando desafios em Kotlin/Triangulo.kt
// Triângulo

/*
Leia 3 valores reais (A, B e C) e verifique se eles formam ou não um triângulo.
Em caso positivo, calcule o perímetro do triângulo e apresente a mensagem:
Perimetro = XX.X

Em caso negativo, calcule a área do trapézio que tem A e B como base e C como
altura, mostrando a mensagem
Area = XX.X

- Entrada

A entrada contém três valores reais.

- Saída

O resultado deve ser apresentado com uma casa decimal.
*/

fun main(args: Array<String>) {
  fun Double.format(digits: Long) = "%.${digits}f".format(this).replace(',','.')
  val (a, b, c) = readLine()!!.split(" ").map {it.toDouble()}

  val p = (a + b + c).format(1)
  val r = (((a + b) * c / 2)).format(1)

  if (a < (b + c) && b < (a + c) && c < (a + b)) println("Perimetro = $p")
  else println("Area = $r")
}

File: JavaScript/Resolvendo Desafios Intermediários em JavaScript/Triangulo.js
// Triângulo

/*
Leia 3 valores reais (A, B e C) e verifique se eles formam ou não um triângulo.
Em caso positivo, calcule o perímetro do triângulo (soma de todos os lados) e
apresente a mensagem:

Perimetro = XX.X

Em caso negativo, calcule a área do trapézio que tem A e B como base e C como
altura, mostrando a mensagem:

Area = XX.X

Fórmula da área de um trapézio: AREA = ((A + B) x C) / 2

- Entrada

A entrada contém três valores reais.

- Saída

O resultado deve ser apresentado com uma casa decimal.
*/

let lines = gets().split('\n')

let line = lines.shift().split(" ");
let A = parseFloat(line[0]);
let B = parseFloat(line[1]);
let C = parseFloat(line[2]);
let maior;
let soma;
let triangulo;

if (A > B && A > C) maior = A;
else if (B > C) maior = B;
else maior = C;

if (maior == A) soma = B + C;
else if (maior == B) soma = A + C;
else soma = B + A;

if (soma > maior) triangulo = true;
else triangulo = false;

if (triangulo) {
  p = A + B + C;
  console.log("Perimetro = ", p.toFixed(1));
} else {
  area = (((A + B) * C) / 2) + .0;
  console.log("Area = ", area.toFixed(1));
}
